chartist.component.ts typescript error when building the UI

**Which plugin/repository is the issue about?**
metrics

**Describe the bug**
When trying to build the admin UI I get:

```
admin-ui/src/extensions/eb6e1994381aaf6a1ded134203bba73449900d8a1578974a2f65ce3958c246ad/chartist/chartist.component.ts(47,35): error TS2345: Argument of type 'undefined[] | ChartEntry' is not assignable to parameter of type 'ChartEntry'.
  Type 'undefined[]' is missing the following properties from type 'ChartEntry': summary, formatOptions
admin-ui/src/extensions/eb6e1994381aaf6a1ded134203bba73449900d8a1578974a2f65ce3958c246ad/chartist/chartist.component.ts(110,53): error TS2345: Argument of type 'undefined[] | ChartEntry' is not assignable to parameter of type 'ChartEntry'.
  Type 'undefined[]' is not assignable to type 'ChartEntry'.
```

**To Reproduce**
Steps to reproduce the behavior:

Using Vendure 3.0.5 and typescript@5.3.3 trying to compile a custom admin UI.

**Expected behavior**
No errors

**Screenshots**
![image](https://github.com/user-attachments/assets/761a961d-1640-4856-b423-4014549299a7)

**Environment**

- NodeJS version - v22.10.0
- Vendure version: 3.0.5
- Plugin version 1.3.1
- Database Postgres but not really applicable
- Any other version numbers from the environment you're using

typescript 5.3.3

**Additional context**

It is possible that there is something unusual about how I'm building my custom UI, but everything else has been working.

--- a/packages/vendure-plugin-metrics/CHANGELOG.md
+++ b/packages/vendure-plugin-metrics/CHANGELOG.md
@@ -1,3 +1,7 @@
+# 1.3.2 (2024-10-29)
+
+- Fix typescript build errors (#525)
+
 # 1.3.1 (2024-08-04)
 
 - Update compatibility range (#480)

--- a/packages/vendure-plugin-metrics/package.json
+++ b/packages/vendure-plugin-metrics/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@pinelab/vendure-plugin-metrics",
-  "version": "1.3.1",
+  "version": "1.3.2",
   "description": "Vendure plugin measuring and visualizing e-commerce metrics",
   "author": "Martijn van de Brug <martijn@pinelab.studio>",
   "homepage": "https://pinelab-plugins.com/",

--- a/packages/vendure-plugin-metrics/src/ui/chartist/chartist.component.ts
+++ b/packages/vendure-plugin-metrics/src/ui/chartist/chartist.component.ts
@@ -33,7 +33,7 @@ export interface ChartEntry {
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ChartistComponent implements OnInit, OnChanges, OnDestroy {
-  @Input() entries: ChartEntry;
+  @Input() entries: ChartEntry | undefined;
   @Input() options?: LineChartOptions = {};
   @ViewChild('chartistDiv', { static: true })
   private chartDivRef: ElementRef<HTMLDivElement>;
@@ -44,7 +44,7 @@ export class ChartistComponent implements OnInit, OnChanges, OnDestroy {
   ngOnInit() {
     this.chart = new LineChart(
       this.chartDivRef.nativeElement,
-      this.entriesToLineChartData(this.entries ?? []),
+      this.entriesToLineChartData(this.entries),
       {
         low: 0,
         // showArea: true,
@@ -89,15 +89,17 @@ export class ChartistComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   formatCurrencyToValue(value: number) {
-    if (this.entries !== undefined) {
+    // Grab a local copy so typescript tracks the truthy check correctly
+    const entries = this.entries;
+    if (entries) {
       const localeFrom =
         localStorage.getItem('vnd__contentLanguageCode') ?? 'en';
       const formatter = new CurrencyPipe(
-        this.entries.formatOptions.locale ?? localeFrom,
-        this.entries.formatOptions.currencyCode
+        entries.formatOptions.locale ?? localeFrom,
+        entries.formatOptions.currencyCode
       );
       const format = (l: number) =>
-        this.entries.formatOptions.formatValueAs === 'currency'
+        entries.formatOptions.formatValueAs === 'currency'
           ? formatter.transform(l) ?? l
           : l;
       return format(value);
@@ -107,7 +109,7 @@ export class ChartistComponent implements OnInit, OnChanges, OnDestroy {
 
   ngOnChanges(changes: SimpleChanges) {
     if ('entries' in changes && this.chart) {
-      this.chart.update(this.entriesToLineChartData(this.entries ?? []));
+      this.chart.update(this.entriesToLineChartData(this.entries));
     }
   }
 
@@ -115,8 +117,8 @@ export class ChartistComponent implements OnInit, OnChanges, OnDestroy {
     this.chart?.detach();
   }
 
-  private entriesToLineChartData(entry: ChartEntry): LineChartData {
-    if (entry.summary?.labels?.length) {
+  private entriesToLineChartData(entry: ChartEntry | undefined): LineChartData {
+    if (entry?.summary.labels?.length) {
       const labels = entry.summary.labels;
       const series = entry.summary.series.map((s) => {
         return s.values.map((v) => {

